### PR TITLE
Async/await refactoring for tests/io folder - test.xpi.js 

### PR DESCRIPTION
**Pull request by hameleonka**
@rpl in this PR I've worked on test.xpi.js in tests/io folder. Please check it out, waiting for your review.

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
Sorry for the hiccup and weird test breakage, https://github.com/mozilla/addons-linter/pull/1749 just got merged and once you rebase your branch to master tests should be working fine again :)

File: tests/io/test.xpi.js [hunk 1]
```diff
@@ -60,25 +60,26 @@ describe('Xpi.open()', function xpiCallback() {
...
     const myXpi = new Xpi('foo/bar', this.fakeZipLib);
     // Return the fake zip to the open callback.
     this.openStub.yieldsAsync(new Error('open() test error'));
-    return myXpi.open()
-      .then(unexpectedSuccess)
-      .catch((err) => {
-        expect(err.message).toContain('open() test');
-      });
+
+    try {
```
Comment (rpl) [inline tests/io/test.xpi.js]: As for #1733, we can use `await expect(...).rejects.toThrow("...")` here (and for the other `try/catch` from this test file).

File: tests/io/test.xpi.js [hunk 2]
```diff
@@ -425,34 +429,33 @@ describe('Xpi.getFileAsStream()', function getFileAsStreamCallback() {
...
-            })
-            .on('end', () => {
-              resolve(chunks);
-            })
-            .on('error', (err) => {
-              reject(err);
-            });
+    const readStream = await myXpi.getFileAsStream('manifest.json');
+
+    const onceReadString = new Promise((resolve, reject) => {
```
Comment (rpl) [inline tests/io/test.xpi.js]: we should be able to reuse the `readStringFromStream` test helper from #1733

File: tests/io/test.xpi.js (hunk 1 above)
Comment (hameleonka) [reply tests/io/test.xpi.js]: @rpl I've changed some Jest methods in this file

File: tests/io/test.xpi.js (hunk 2 above)
Comment (hameleonka) [reply tests/io/test.xpi.js]: @rpl I can't make this change until #1733 is merged (because there I included readStringFromStream into helpers)

Comment (hameleonka) [reply tests/io/test.xpi.js]: @rpl I've made necessary changes. Please check it out.

**Comment by EnTeQuAk** [comment]
Glorious, merged! 👍

**Pull request merged**